src/core/KeyPair.ts

```diff
@@ -34,7 +34,7 @@ export abstract class KeyPair {
     /**
      * Abstract method to derive public key from private key
      */
-    public abstract getPublicKey(): Key;
+    protected abstract getPublicKey(): Key;
```

Comment:
i would rename this derivePublicKey to make clear it is not a simple accessor

Reply:
I agree. The double get method got me confused.

I had a second look, my opinion is not a good practice to call an instance method in a constructor. The object being call may not be fully created yet (not in this case but in general this could be an issue). I have changed the method to be private static for each subclass.

Reply:
agree to that.
> my opinion is not a good practice to call an instance method in a constructor. The object being called may not be fully created yet (not in this case but in general this could be an issue).